lib/scala/profiling-utils/src/main/java/org/enso/profiling/sampler/OutputStreamSampler.java

```diff
@@ -0,0 +1,56 @@
+package org.enso.profiling.sampler;
+
+import java.io.DataOutputStream;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import org.netbeans.modules.sampler.Sampler;
+
+/**
+ * Gathers application performance statistics that can be visualised in Java VisualVM, and writes it
+ * to the provided output.
+ */
+public final class OutputStreamSampler implements MethodsSampler {
```

Comment:
To keep the API's _conceptual surface_ smaller, I'd suggest to make these implementations package private and only expose `MethodsSampler`.